src/lib/api.ts

```diff
@@ -11,17 +24,26 @@ export class UninitializedApiError extends Error {
 
 export class AeriesApi {
 	public apiUrl: URL;
-	private token: string | null;
-	private student: Student | null;
+	public authedStudent: Writable<AuthedStudent | null>;
 
-	public constructor(apiUrl: URL, token: string | null, student: Student | null) {
+	public constructor(apiUrl: URL, authedStudent: AuthedStudent | null) {
 		this.apiUrl = apiUrl;
-		this.token = token;
-		this.student = student;
+		this.authedStudent = writable(authedStudent);
+	}
+
+	public isInitialized(): boolean {
+		return get(this.authedStudent) != null;
 	}
 
 	// return true if able to authenticate
-	public async authenticate(username: string, password: string): Promise<Boolean> {
+	public async authenticate(
+		username: string,
+		password: string
+	): Promise<Boolean> {
+		if (get(this.authedStudent) != null) {
```

Comment:
isn't this the exact same as the body of `isIntialized`? If this is because of a type error, I believe there is a way to tell typescript that this is a type guard.

Reply:
You are correct, I wrote that before I added isInitialized today